zebra-rpc/src/methods/get_block_template_rpcs/types/get_block_template_opts.rs

```diff
@@ -0,0 +1,91 @@
+//! Parameter types for the `getblocktemplate` RPC.
+
+use super::hex_data::HexData;
+
+/// Defines whether the RPC method should generate a block template or attempt to validate a block proposal.
+/// `Proposal` mode is currently unsupported and will return an error.
+#[derive(Debug, serde::Deserialize, PartialEq, Eq)]
+#[serde(rename_all = "lowercase")]
+pub enum GetBlockTemplateRequestMode {
+    /// Indicates a request for a block template.
+    Template,
+
+    /// Indicates a request to validate block data.
+    /// Currently unsupported and will return an error.
+    Proposal,
+}
+
+impl Default for GetBlockTemplateRequestMode {
+    fn default() -> Self {
+        Self::Template
+    }
+}
+
+/// Valid `capabilities` values that indicate client-side support.
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "lowercase")]
+#[allow(dead_code)]
+pub enum GetBlockTemplateCapability {
+    /// Long Polling support.
+    /// Currently ignored by zebra.
+    LongPoll,
+
+    /// Information for coinbase transaction, default template data with the `coinbasetxn` field.
+    /// Currently ignored by zebra.
+    CoinbaseTxn,
+
+    /// Coinbase value, template response provides a `coinbasevalue` field and omits `coinbasetxn` field.
+    /// Currently ignored by zebra.
+    CoinbaseValue,
+
+    /// Components of the coinbase transaction.
+    /// Currently ignored by zebra.
+    CoinbaseAux,
+
+    /// Currently ignored by zcashd and zebra.
+    Proposal,
+
+    /// Currently ignored by zcashd and zebra.
+    ServerList,
+
+    /// Currently ignored by zcashd and zebra.
+    WorkId,
+
+    /// Mutations
+    // TODO: Fill out valid mutations capabilities.
+    UnknownCapability(String),
+}
```

Comment:
I think this can work using `#[serde(other)]`, but I'm not sure of the exact syntax:
https://serde.rs/variant-attrs.html#other
https://serde.rs/enum-representations.html#internally-tagged

```suggestion
#[derive(Clone, Debug, PartialEq, Eq, serde::Deserialize)]
#[serde(tag = "type", rename_all = "lowercase")]
pub enum GetBlockTemplateCapability {
    /// Long Polling support.
    /// Currently ignored by zebra.
    LongPoll,

    /// Information for coinbase transaction, default template data with the `coinbasetxn` field.
    /// Currently ignored by zebra.
    CoinbaseTxn,

    /// Coinbase value, template response provides a `coinbasevalue` field and omits `coinbasetxn` field.
    /// Currently ignored by zebra.
    CoinbaseValue,

    /// Components of the coinbase transaction.
    /// Currently ignored by zebra.
    CoinbaseAux,

    /// Currently ignored by zcashd and zebra.
    Proposal,

    /// Currently ignored by zcashd and zebra.
    ServerList,

    /// Currently ignored by zcashd and zebra.
    WorkId,

    /// Mutations
    #[serde(other)]
    UnknownCapability(String),
}
```

Reply:
Yes!!! 💯 It just has to a unit variant. I applied a slightly modified version of this.

Reply:
Yes!!! 💯 It just has to be a unit variant. I applied a slightly modified version of this.